server/rest/elasticsearch_source.py

```diff
@@ -0,0 +1,133 @@
+                'index': index,
+                'base_url': baseURL,
+                'host_name': hostName
+            }
+        }
+        if username and password:
+            enc_creds = encryptCredentials("{}:{}".format(
+                username, password))
+            minerva_metadata['elasticsearch_params']['credentials'] = enc_creds
+        desc = 'elasticsearch source for  %s' % name
+        return self.createSource(name, minerva_metadata, desc)
+    createElasticsearchSource.description = (
+        Description('Create a source from an external elasticsearch server.')
+        .responseClass('Item')
+        .param('name', 'The name of the elasticsearch source', required=True)
+        .param('baseURL', 'URL of the elasticsearch service', required=True)
+        .param('index', 'Index of interest', required=True)
+        .param('username', 'Elasticsearch username', required=False)
+        .param('password', 'Elasticsearch password', required=False)
+        .errorResponse('Write permission denied on the source folder.', 403))
+
+
+class ElasticsearchQuery(Resource):
+
+    def __init__(self):
+        self.resourceName = 'minerva_query_elasticsearch'
+        self.route('POST', (), self.queryElasticsearch)
+
+    @access.user
+    def queryElasticsearch(self, params):
+        """
+        Creates a dataset to store the results for an elastic search query,
+        then calls a local job running the elasticsearch_worker.
+        """
+        currentUser = self.getCurrentUser()
+        datasetName = params['datasetName']
+        elasticsearchParams = params['searchParams']
+
+        datasetFolder = findDatasetFolder(currentUser, currentUser)
+        dataset = (self.model('item').createItem(
+            datasetName,
+            currentUser,
+            datasetFolder,
+            'created by elasticsearch query'))
+
+        user, token = self.getCurrentUser(returnToken=True)
+        kwargs = {
+            'params': params,
+            'user': currentUser,
+            'dataset': dataset,
+            'token': token,
+            'sourceId': params['sourceId']
+        }
+
+        job = self.model('job', 'jobs').createLocalJob(
+            title='elasticsearch: %s' % datasetName,
+            user=currentUser,
+            type='elasticsearch',
+            public=False,
+            kwargs=kwargs,
+            module='girder.plugins.minerva.jobs.elasticsearch_worker',
+            async=True)
+
+        if 'meta' in dataset:
+            metadata = dataset['meta']
+        else:
+            metadata = {}
+
+        minerva_metadata = {
+            'dataset_id': dataset['_id'],
+            'source': 'elasticsearch',
+            'elasticsearch_params': elasticsearchParams,
+            'original_type': 'json'
+        }
+        metadata['minerva'] = minerva_metadata
+        self.model('item').setMetadata(dataset, metadata)
+
+        self.model('job', 'jobs').scheduleJob(job)
+
+        return minerva_metadata
```

Comment:
can you `return job` instead of `return minerva_metadata` ?

What I'm trying to push towards is a convention where if some async job is created by an endpoint, the endpoint returns the job.